src/cli/api/local_api.py

```diff
@@ -733,20 +713,189 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+                        app_schema_path = Path(os.path.join(BASE_DIR), 'schema', 'applications', '')
+                        valid, message = validate_json_schema(experiment_input_path / file, app_schema_path / file)
+                        if valid:
+                            if file == 'network.json':
+                                # Check if the app files for the roles from network.json exist in the
+                                # experiment/input directory
+                                application_file_names = self.__get_role_file_names(experiment_input_path)
+                                for application_file_name in application_file_names:
+                                    if not (experiment_input_path / application_file_name).is_file():
+                                        error_dict["error"].append(f"'{experiment_input_path}' is missing the file: "
+                                                                   f"'{application_file_name}'")
+                        else:
+                            error_dict["error"].append(message)
+                    else:
+                        error_dict["error"].append(f"'{experiment_input_path}' should contain the file '{file}'")
+        else:
+            error_dict["error"].append(f"Required directory not found: '{experiment_input_path}'")
+
+    def _validate_experiment_nodes(self, experiment_file_path: Path, experiment_data: Dict[str, Any], error_dict:
+                                  ErrorDictType) -> None:
+        """
+        Validate if the amount of nodes (defined in the experiment.json file) are valid for 'network_slug' and if
+        all the nodes exist and belong to this network.
+
+        Args:
+            experiment_file_path: The location of the experiment.json file
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+        experiment_network_slug = experiment_data["asset"]["network"]["slug"]
+
+        # Check if the amount of nodes are valid for this network
+        experiment_nodes = experiment_data["asset"]["network"]["nodes"]
+
+        all_network_nodes = self._get_network_nodes()
+        network_nodes = all_network_nodes[experiment_network_slug]
+        if len(experiment_nodes) > len(network_nodes):
+            error_dict["error"].append(f"In file '{experiment_file_path}': too many nodes used in network "
+                                       f"'{experiment_network_slug}'. Maximum amount of nodes that can be used: "
+                                       f"{len(network_nodes)}")
+
+        # Check if all nodes exist and belong to this network
+        for node in experiment_nodes:
+            if node["slug"] not in network_nodes:
+                error_dict["error"].append(f"In file '{experiment_file_path}': '{node['slug']}' does not exist or does "
```

Comment:
Node '{node['slug']}' does not exist 
This would make the error message more clear about what does not exist.

Reply:
Will add it